test/back_chaining.lean

```diff
@@ -2,6 +2,10 @@ open tactic
 
 variables {α : Type*}
 
+instance : has_subset (set α) := ⟨λ s t, ∀ ⦃x⦄, x ∈ s → x ∈ t⟩
+instance : has_union (set α) := ⟨λ s t, {a | a ∈ s ∨ a ∈ t}⟩
+instance : has_inter (set α) := ⟨λ s t, {a | a ∈ s ∧ a ∈ t}⟩
```

Comment:
An alternative would be to `import data.set.basic` instead, but I am not familiar with tactic tests enough to know if that's appropriate. The same comment applies to the change in `test/rcases.lean`.

Reply:
Importing very low-level files is fine, but not importing anything is probably nicer.